### ISSUE TITLE
llvm should be built with zlib support as lld needs it

### ISSUE THREAD

**Issue by ifreund** [ziglang/zig-bootstrap#57]
currently we disable zlib support here:
https://github.com/ziglang/zig-bootstrap/blob/fc78f1770c0bccc5262577456da4172ae36abbc2/build#L66

However, this may be necessary when linking against system libraries including libc, which current fails on void linux using the static zig tarballs from ziglang.org/download which are built using this repo:
```
ld.lld: error: /usr/lib64/gcc/x86_64-unknown-linux-gnu/10.2.1/../../../../lib64/crti.o: contains a compressed section, but zlib is not available
```

To fix this I guess we need to build zlib as well as part of this repo, I don't really see an alternative option unfortunately.

**Comment by name-changer** [comment]
Hi! I'm one of the people having problems with this. Since I updated to the latest zig master a few days, I get the following errors when I try to build my (hobby-gamedev-)project:

```
These are all repeated several times:
ld.lld: error: /usr/lib64/gcc/x86_64-suse-linux/10/../../../../lib64/crti.o: contains a compressed section, but zlib is not available 
ld.lld: error: /usr/lib64/libc_nonshared.a(elf-init.oS): contains a compressed section, but zlib is not available
ld.lld: error: /usr/lib64/libc_nonshared.a(atexit.oS): contains a compressed section, but zlib is not available
ld.lld: error: /usr/lib64/libc_nonshared.a(pthread_atfork.oS): contains a compressed section, but zlib is not available
ld.lld: error: /usr/lib64/gcc/x86_64-suse-linux/10/../../../../lib64/crtn.o: contains a compressed section, but zlib is not available
error: LLDReportedFailure
```

My OS is the latest OpenSuse Tumbleweed (it's a rolling distro, so always quite up-to-date), and the project uses libc, SDL, OpenGL, stb_image, dear imgui.   

I post this, because I have two questions: are there any workarounds I could use until this issue is adressed? 

And out of curiosity: I wonder, shouldn't there be a lot of people having problems with this? Or is it so uncommon that libc uses compressed sections that this problem doesn't appear that often? 

Thanks!

**Comment by ifreund** [comment]
> I post this, because I have two questions: are there any workarounds I could use until this issue is adressed?

As a workaround, using `native-linux-gnu` or `native-linux-musl` as the target will cause zig to build and use its own libc instead of the system libc. This should circumvent this issue but may cause others (e.g. if your system uses a newer glibc version than zig can provide).

> And out of curiosity: I wonder, shouldn't there be a lot of people having problems with this? Or is it so uncommon that libc uses compressed sections that this problem doesn't appear that often?

Yes, I expect this will affect most people using the zig tarballs from the website and linking system libraries, it should be fixed before 0.8.0 releases.

**Comment by name-changer** [comment]
Thanks for the reply! I tried the two targets, but as you expected it caused other problems with the cImports. 

Since building zig from source is also a problem with OpenSuse Tumbleweed (lld-devel is not available), I guess I'll just have to wait a bit with the gamedev stuff, and use zig without C for now.
